### DISCUSSION TITLE
Is there a version of OpenRoad that does not use dynamic link libraries？

### DISCUSSION THREAD

**Discussion by inv1s10n** [The-OpenROAD-Project/OpenROAD#6110] [Q&A] [answered]
How to compile an openroad version that does not use dynamic link libraries? I need to use openroad in a place where I cannot download library files.

**Comment by maliberty** [comment, marked as answer]
OR itself uses static libraries but system dependencies are whatever are provided.  You can always copy the shared libraries as needed.  Docker is another option.

**Reply by inv1s10n** [reply]
I tried copy OR's need (all .so flies are from cmd "ldd openroad"). However after I setenv LD_LIBRARY_PATH (in csh), except for one .so file, all other .so files can be found by OR normally, and these .so files are in the same directory. I don't know where the problem lies.
The error log : 
./openroad: error while loading shared libraries: libQt5Core.so.5: cannot open shared object file: No such file or directory

**Reply by maliberty** [reply]
There is nothing special about that file, I assume you didn't copy it.

**Reply by inv1s10n** [reply]
![image](https://github.com/user-attachments/assets/4dee823e-5b5f-4e5d-bc11-f78ed3fb2ccb)
These .so files are all here, a same dir.

**Reply by maliberty** [reply]
Dunno.  Perhaps your LD_LIBRARY_PATH has an issue or perhaps the libraries are not compatible across the machines.
